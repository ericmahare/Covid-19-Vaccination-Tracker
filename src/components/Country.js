import { Link } from "react-router-dom";
const Country = ({ info }) => {
  const {administered, country, abbreviation} = info
  return (
      <div className="country-card">
    <Link to={`/${country}`}>
      <img src={`https://countryflagsapi.com/png/${abbreviation}`} alt='country-flag' className='country-icon'/>
      <div className='card-text'>
        <h2>{country}</h2>
        <h2>{administered.toLocaleString()}</h2>
      </div>
      <div className='card-arrow'>
          <p><i className="fa-solid fa-circle-arrow-right"></i></p>
      </div>
    </Link>
      </div>
  )
}
export default Country;