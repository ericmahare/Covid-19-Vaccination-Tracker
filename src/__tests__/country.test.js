import { BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Country from '../components/Country';

const info = {
  administered: 23000,
  country: 'Kenya',
  abbreviation: 'KE'
}

it('renders properly', () => {
  const tree = renderer.create(
    <Router>
      <Country info={info}/>
    </Router>
  ).toJSON()
  expect(tree).toMatchSnapshot();
})