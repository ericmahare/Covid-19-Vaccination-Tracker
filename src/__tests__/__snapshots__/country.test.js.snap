// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="country-card"
>
  <a
    href="/Kenya"
    onClick={[Function]}
  >
    <img
      alt="country-flag"
      className="country-icon"
      src="https://countryflagsapi.com/png/KE"
    />
    <div
      className="card-text"
    >
      <h2>
        Kenya
      </h2>
      <h2>
        23,000
      </h2>
    </div>
    <div
      className="card-arrow"
    >
      <p>
        <i
          className="fa-solid fa-circle-arrow-right"
        />
      </p>
    </div>
  </a>
</div>
`;
